Lib/test/test_threading.py

```diff
@@ -1827,6 +1827,39 @@ def test__all__(self):
         support.check__all__(self, threading, ('threading', '_thread'),
                              extra=extra, not_exported=not_exported)
 
+    @requires_subprocess()
+    def test_gh112826_missing__thread__is_main_interpreter(self):
+        with os_helper.temp_dir() as tempdir:
+            modname = '_thread_fake'
+            import os.path
+            filename = os.path.join(tempdir, modname + '.py')
+            with open(filename, 'w') as outfile:
+                outfile.write("""if True:
+                    import _thread
+                    ns = globals().update(vars(_thread))
+                    #from _thread import *
+                    del _is_main_interpreter
+                    """)
+            expected_output = 'success!'
+            script = f"""if True:
+                import sys
+                sys.path.insert(0, {tempdir!r})
+                import {modname}
+                sys.modules['_thread'] = _thread_fake
```

Comment:
```suggestion
                sys.modules['_thread'] = {modname}
```

You should decide between injecting modname or hardcoding _thread_fake, but mixing them is not a good idea :)

Reply:
fixed